src/main/java/com/example/provider/VehicleServlet.java

```diff
@@ -220,19 +231,10 @@ public void doPut(HttpServletRequest request, HttpServletResponse response) thro
 
     try {
       updatedVehicle = vehicleServiceClient.updateVehicle(updateVehicleRequest);
-    } catch (StatusRuntimeException e) {
-      logger.warning(String.format("GRPC reported exception: %s", e.getStatus()));
-
-      if (e.getStatus().getCode().equals(Status.NOT_FOUND.getCode())) {
-        response.sendError(
-            HttpServletResponse.SC_BAD_REQUEST,
-            String.format("Vehicle with Id: %s does not exist.", vehicleId));
-        return;
-      }
-
+    } catch (NotFoundException e) {
```

Comment:
is it possible to have any other exceptions such as `StatusRuntimeException` here? Wonder if we should catch those exceptions as well so the functionality of reporting an internal server error stays the same for non NotFoundExceptions

Reply:
Hey Shirwa! this would actually still let the other exceptions flow and surface as InternalServerErrors like before. But I do think we should catch other types of exceptions here (401) and surface them properly. I'll open a task to get there